### DISCUSSION TITLE
"How to prevent game from restarting on screen size change in Forge2D?"

### DISCUSSION THREAD

**Discussion by ivanrodriguezromero1** [flame-engine/forge2d#77] [General]
I'm developing a game using the Forge2D game engine in Flutter, but every time the screen size changes (e.g. rotating the device), the game restarts from the beginning. I would like to prevent this behavior and instead maintain the current state of the game across screen size changes. How can I accomplish this? I've tried implementing the resize method and using a Viewport widget, but I'm still experiencing the issue. Any help or guidance would be greatly appreciated. Thanks!

**Comment by spydon** [comment]
Hi, use `GameWidget.controlled` instead of the normal `GameWidget` constructor.

**Reply by ivanrodriguezromero1** [reply]
Thank you very much for your help! Your suggestion to use GameWidget.controlled instead of the normal GameWidget constructor solved my problem. I really appreciate it.
